### PR TITLE
CXP-658: Add marketplace page

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

Screenshot_2021-04-23_16-50-06

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import {Connect} from '@akeneo-pim-community/connectivity-connection';
+import React from 'react';
+import {dependencies} from '../dependencies';
+import highlightMenu from '../menu/highlight-menu';
+import ReactController from '../react/react-controller';
+
+class ConnectController extends ReactController {
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx:7]: Why do you have a new Controller here ?

Comment (tseho) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx:7]: Ok, I get it now.
Both "Data flows" & "Connection settings" still have their own controllers.
I would rather have a new Controller "MarketplaceController"

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  reactElementToMount() {
+    return <Connect dependencies={dependencies} />;
+  }
+
+  routeGuardToUnmount() {
+    return /^akeneo_connectivity_connection_connect/;
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx:13]:
```suggestion
    return /^akeneo_connectivity_connection_connect_marketplace/;
```

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/MarketplaceSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, {StrictMode} from 'react';
+import {HashRouter as Router} from 'react-router-dom';
+import {AkeneoThemeProvider} from './akeneo-theme-provider';
+import {withDependencies} from './dependencies-provider';
+import {Marketplace} from '../connect/pages/Marketplace';
+
+export const MarketplaceSettings = withDependencies(() => (
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/infrastructure/MarketplaceSettings.tsx:7]:
```suggestion
export const Marketplace = withDependencies(() => (
```
This should probably just be "Marketplace" (seems like a remnant of the "ConnectionSettings" app)

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx (hunk 1 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/js/controller/connect.tsx]: Both naming seems ok to me, but if we use "Connect" right now it implies that we are going to migrate the others controllers/apps inside this one.

**Review by tseho — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/front/src/connect/pages/Marketplace.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    & > * {
+        margin-bottom: 20px;
+    }
+`;
+
+export const Marketplace: FC = () => {
+    const translate = useTranslate();
+    const breadcrumb = (
+        <Breadcrumb>
+            <Breadcrumb.Step>{translate('pim_menu.tab.connect')}</Breadcrumb.Step>
```
[the remark this answers is not in the document]
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/front/src/connect/pages/Marketplace.tsx:71]: No need, I'm already on the connect menu IMHO.

[the remark this answers is not in the document]
Comment (RiffVentura) [reply src/Akeneo/Connectivity/Connection/front/src/connect/pages/Marketplace.tsx:71]: Fine by me (although it slightly breaks consistency with other breadcrumbs)

**Review by pchasle — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**